policies/templates/gcp_enforce_labels_v1.yaml

```diff
@@ -26,79 +26,125 @@ spec:
-              
-              # get_label for standard resources
-              # placeholder function: in case we run into non-standard location for labels
-              
-              get_label(asset) = resource {
-              	resource_types_to_scan[_] == asset.asset_type
-              	resource := get_generic_label(asset)
-              }
-              #ENDINLINE
\ No newline at end of file
+       #
+       # Copyright 2018 Google LLC
+       #
+       # Licensed under the Apache License, Version 2.0 (the "License");
+       # you may not use this file except in compliance with the License.
+       # You may obtain a copy of the License at
+       #
+       #      http://www.apache.org/licenses/LICENSE-2.0
+       #
+       # Unless required by applicable law or agreed to in writing, software
+       # distributed under the License is distributed on an "AS IS" BASIS,
+       # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+       # See the License for the specific language governing permissions and
+       # limitations under the License.
+       #
+       
+       package templates.gcp.GCPEnforceLabelConstraintV1
+       
+       import data.validator.gcp.lib as lib
+       
+       deny[{
+       	"msg": message,
+       	"details": metadata,
+       }] {
+       	constraint := input.constraint
+       	lib.get_constraint_params(constraint, params)
+       	asset := input.asset
+       
+       	mandatory_label := params.mandatory_labels[_]
+       	label_value_pattern := mandatory_label[label_key]
+       
+       	tested_resource_types := [
+       		"cloudresourcemanager.googleapis.com/Project",
+       		"storage.googleapis.com/Bucket",
+       		"compute.googleapis.com/Instance",
+       		"compute.googleapis.com/Image",
+       		"compute.googleapis.com/Disk",
+       		"compute.googleapis.com/Snapshot",
+       		"google.bigtable.Instance",
+       		"sqladmin.googleapis.com/Instance",
+       	]
+       
+       	standard_types := [
+       		"cloudresourcemanager.googleapis.com/Project",
+       		"storage.googleapis.com/Bucket",
+       		"compute.googleapis.com/Instance",
+       		"compute.googleapis.com/Image",
+       		"compute.googleapis.com/Disk",
+       		"compute.googleapis.com/Snapshot",
+       		"google.bigtable.Instance",
+       	]
+       
+       	resource_types_to_scan := lib.get_default(params, "resource_types_to_scan", tested_resource_types)
+       
+       	resource_is_to_scan(asset, resource_types_to_scan)
+       
+       	not label_is_valid(label_key, label_value_pattern, asset, standard_types)
+       
+       	message := sprintf("%v's label is in violation.", [asset.name])
+       	metadata := {"resource": asset.name, "label_in_violation": label_key}
+       }
+       
+       resource_is_to_scan(asset, resource_types_to_scan) {
+       	resource_types_to_scan[_] == asset.asset_type
+       }
+       
+       # check if label exists for all resources to scan
```

Comment:
nit: this comment isn't exactly accurate, since we're checking for label value in addition to existence

Reply:
true, I forgot to update this one. Wil lfix in next commit